MapboxCoreNavigation/RouteController.swift

```diff
@@ -236,20 +236,30 @@ extension RouteController: CLLocationManagerDelegate {
             if !isCloseToCurrentStep && isCloseToUpComingStep {
                 
                 // Increment the step
-                routeProgress.currentLegProgress.stepIndex += 1
```

Comment:
So did this turn out to be a good change to make? We’re going to snap to the maneuver at the _start_ of the current step before incrementing the step. `userSnapToStepDistanceFromManeuver` could be as long as the step itself, and `secondsToEndOfStep` could be as long as the step’s duration.